packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -757,6 +727,96 @@ class IOSDevice extends Device {
     _logger.printError('');
   }
 
+  /// Find the Dart VM url using ProtocolDiscovery (logs from `idevicesyslog`)
+  /// and mDNS simultaneously, using whichever is found first. `idevicesyslog`
+  /// does not work on wireless devices, so only use mDNS for wireless devices.
+  /// Wireless devices require using the device IP as the host.
+  Future<Uri?> _discoverDartVMForCoreDevice({
+    required String packageId,
+    required bool ipv6,
+    required DebuggingOptions debuggingOptions,
+    ProtocolDiscovery? vmServiceDiscovery,
+  }) async {
+    Timer? maxWaitForCI;
+    final Completer<Uri?> cancelCompleter = Completer<Uri?>();
+
+    // When testing in CI, wait a max of 10 minutes for the Dart VM to be found.
+    // Afterwards, stop the app from running and upload DerivedData Logs to debug
+    // logs directory. CoreDevices are run through Xcode and launch logs are
+    // therefore found in DerivedData.
+    if (debuggingOptions.usingCISystem && debuggingOptions.debugLogsDirectoryPath != null) {
+      maxWaitForCI = Timer(const Duration(minutes: 10), () async {
+        _logger.printError('Failed to find Dart VM after 10 minutes.');
+        await _xcodeDebug.exit();
+        final String? homePath = _platform.environment['HOME'];
+        Directory? derivedData;
+        if (homePath != null) {
+          derivedData = _fileSystem.directory(
+            _fileSystem.path.join(homePath, 'Library', 'Developer', 'Xcode', 'DerivedData'),
+          );
+        }
+        if (derivedData != null && derivedData.existsSync()) {
+          final Directory debugLogsDirectory = _fileSystem.directory(
+            debuggingOptions.debugLogsDirectoryPath,
+          );
+          debugLogsDirectory.createSync(recursive: true);
+          for (final FileSystemEntity entity in derivedData.listSync()) {
+            if (entity is! Directory || !entity.childDirectory('Logs').existsSync()) {
+              continue;
+            }
+            final Directory logsToCopy = entity.childDirectory('Logs');
+            final Directory copyDestination = debugLogsDirectory
+                .childDirectory('DerivedDataLogs')
+                .childDirectory(entity.basename)
+                .childDirectory('Logs');
+            _logger.printTrace('Copying logs ${logsToCopy.path} to ${copyDestination.path}...');
+            copyDirectory(logsToCopy, copyDestination);
```

Comment:
You may wind up needing to compress this before you copy it if you hit this problem https://github.com/flutter/flutter/pull/75736 :

![Screen Shot 2021-02-09 at 1 00 49 PM](https://user-images.githubusercontent.com/682784/107434291-72ba5f00-6adf-11eb-9880-e27cd1592e35.png)

I'm not sure how logs are handled these days, may as well merge this and see.

Reply:
yolo